pylint/checkers/utils.py

```diff
@@ -1457,11 +1457,15 @@ def is_registered_in_singledispatch_function(node: nodes.FunctionDef) -> bool:
 
     decorators = node.decorators.nodes if node.decorators else []
     for decorator in decorators:
-        # func.register are function calls
-        if not isinstance(decorator, nodes.Call):
+        # func.register are function calls or register attributes
+        # when the function is annotated with types
+        if isinstance(decorator, nodes.Call):
+            func = decorator.func
+        elif isinstance(decorator, nodes.Attribute):
+            func = decorator
+        else:
```

Comment:
Still needed to adjust this part cause we should consider, for..
```python
@singledispatch
def example(arg):
    pass
```

..that both below are classified as possible registers:
```python
@example.register(str)
def _(arg):
    pass

@example.register
def _(arg: str):
    pass
```

Reply:
Do you need any help with that?